### PR TITLE
Package dependencies

### PR DESCRIPTION
**Pull request by saulfield**
This PR makes progress towards #1981.

Number of dependencies for a base install of smarts with no extras (before -> after):
- Required packages: 24 -> 17
- Total installed packages: 57 -> 28

An `[all]` extra has been added which installs all extras. A CI test to check that the `[all]` install succeeds with no conflicts has also been added.

## Removed
- `pandas`
- `rich`
- `sh`

## Moved out of core
- `tornado`
- `websocket-client`
- `ijson`
- `matplotlib`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,10 +13,13 @@ Copy and pasting the git commit messages is __NOT__ enough.
 ### Changed
 - Changed waypoints in sumo maps to use more incoming lanes into junctions.
 - Increased the cutoff radius for filtering out waypoints that are too far away in junctions in sumo maps.
+- Made Envision dependencies optional. Use `pip install -e .[envision]` to install them.
```
Comment (Gamenot) [inline CHANGELOG.md:16]: Envision does not appear to be optional since files like `agent_manager.py` appear to still import from envision.

**Review by Gamenot — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-format.yml [hunk 2]
```diff
@@ -94,7 +94,7 @@ jobs:
           . ${{env.venv_dir}}/bin/activate
           pip install --upgrade pip "setuptools<58.3.0"
           pip install wheel==0.38.4
-          pip install .[camera_obs,doc,train,ray]
+          pip install .[camera_obs,doc,train,ray,envision,argoverse,opendrive,waymo]
```
Comment (Adaickalavan) [inline .github/workflows/ci-format.yml:97]: In doc test, given we have `autodoc_mock_imports`, are the additional installs envision, argoverse, opendrive, waymo necessary?

Comment (saulfield) [reply .github/workflows/ci-format.yml:97]: Yes, see https://github.com/huawei-noah/SMARTS/actions/runs/4875402139/jobs/8697543728#step:5:847

File: smarts/core/waymo_map.py [hunk 3]
```diff
@@ -47,19 +59,12 @@
...
-    line_intersect_vectorized,
-    radians_to_vec,
-    ray_boundary_intersect,
-    vec_2d,
-)
+try:
+    import rtree
+except:
+    raise ImportError(
+        "You may not have installed the [argoverse] dependencies required for using Argoverse 2 maps with SMARTS. Install it first using the command `pip install -e .[argoverse]` at the source directory."
```
Comment (Adaickalavan) [inline smarts/core/waymo_map.py:66]: Error message indicates Argoverse 2 maps, but file is waymo_map.py.

File: smarts/waymo/waymo_open_dataset/protos/__init__.py [hunk 4]
```diff
@@ -12,3 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+
+try:
+    import google.protobuf
+except:
+    raise ImportError(
+        "You may not have installed the [waymo] dependencies required for using Argoverse 2 maps with SMARTS. Install it first using the command `pip install -e .[waymo]` at the source directory."
+    )
```
Comment (Adaickalavan) [inline smarts/waymo/waymo_open_dataset/protos/__init__.py:21]: Error message indicates Argoverse 2 maps, but folder is waymo_open_dataset.

**Review by Adaickalavan — APPROVED** [review]

File: smarts/core/waymo_map.py (hunk 3 above)
Comment (saulfield) [reply smarts/core/waymo_map.py]: Good catch, fixed.

File: smarts/waymo/waymo_open_dataset/protos/__init__.py (hunk 4 above)
Comment (saulfield) [reply smarts/waymo/waymo_open_dataset/protos/__init__.py]: Thanks, fixed.

File: CHANGELOG.md (hunk 1 above)
Comment (Gamenot) [reply CHANGELOG.md:17]: OK.

**Review by Gamenot — APPROVED** [review]

**Pull request merged**